src/rpc/rawtransaction.cpp

```diff
@@ -822,20 +822,23 @@ UniValue signrawtransaction(const JSONRPCRequest& request)
                 view.AddCoin(out, std::move(newcoin), true);
             }
 
-            // if redeemScript given and not using the local wallet (private keys
-            // given), add redeemScript to the tempKeystore so it can be signed:
-            if (fGivenKeys && (scriptPubKey.IsPayToScriptHash() || scriptPubKey.IsPayToWitnessScriptHash())) {
-                RPCTypeCheckObj(prevOut,
-                    {
-                        {"txid", UniValueType(UniValue::VSTR)},
-                        {"vout", UniValueType(UniValue::VNUM)},
-                        {"scriptPubKey", UniValueType(UniValue::VSTR)},
-                        {"redeemScript", UniValueType(UniValue::VSTR)},
-                    });
-                UniValue v = find_value(prevOut, "redeemScript");
-                if (!v.isNull()) {
-                    std::vector<unsigned char> rsData(ParseHexV(v, "redeemScript"));
-                    CScript redeemScript(rsData.begin(), rsData.end());
+            // if redeemScript given add redeemScript to the tempKeystore so it can be signed:
+            if (scriptPubKey.IsPayToScriptHash() || scriptPubKey.IsPayToWitnessScriptHash()) {
+                UniValue rs = find_value(prevOut, "redeemScript");
+                if (!fGivenKeys && rs != NullUniValue) {
+                    // if private keys aren't given, assume we are using the local wallet, so we should not accept scripts
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "Scripts cannot be provided when using local wallet");  
+                }
+                std::vector<UniValue> scripts;
+                if (rs.getType() == UniValue::VSTR) {
+                    scripts.push_back(rs);
+                } else if (rs.getType() == UniValue::VARR) {
+                    // If redeemScript is an array, iterate over all scripts provided
+                    scripts = rs.getValues();
+                }
```

Comment:
Missing error when not string or array? Suggestion (`scripts` is a vector of strings):
```cpp
if (rs.getType() == UniValue::VARR) {
    for (auto r : rs.getValues()) {
         scripts.push_back(r.get_str());
    }
} else {
    scripts.push_back(rs.get_str());
}
```

Reply:
Nice idea, done, thanks :)